javaslang/src/main/java/javaslang/collection/Arrays.java

```diff
@@ -35,6 +39,19 @@
         return array;
     }
 
+    @SuppressWarnings("unchecked")
+    static <T> Object[] asArray(Iterable<? extends T> iterable) {
+        if (iterable instanceof Collection<?>) {
+            final Collection<? extends T> collection = (Collection<? extends T>) iterable;
+            return collection.toArray();
+        } else if (iterable instanceof Traversable) {
+            final Traversable<T> traversable = asTraversableAgain(iterable);
```

Comment:
It is, but where do we convert it to a `TraversableAgain`, if you want to delete the `asTraversableAgain` method? If I just igore those, I need to fix the tests, as I've done for `Euler67Test`